pkg/cloud/network.go

```diff
@@ -86,6 +115,32 @@ func (c *client) GetOrCreateNetwork(csCluster *infrav1.CloudStackCluster) (retEr
 	csCluster.Status.NetworkID = resp.Id
 	csCluster.Status.NetworkType = resp.Type
 
+	return tagNetwork(csCluster, true)
+}
+
+func (c *client) UntagAndDestroyNetwork(csCluster *infrav1.CloudStackCluster) (retError error) {
```

Comment:
I think I'd prefer these to be two separate methods. An untag and then a deleteIfNotInUse or something like that.

Reply:
That would make the intent clearer.  Good idea!